src/main/java/de/tum/in/www1/artemis/repository/specs/MessageSpecs.java

```diff
@@ -25,14 +26,7 @@ public class MessageSpecs {
      * @return specification used to chain DB operations
      */
     public static Specification<Post> getConversationSpecification(Long conversationId) {
-        return ((root, query, criteriaBuilder) -> {
-            // fetch additional values to avoid subsequent db calls
-            var answerFetch = root.fetch(Post_.ANSWERS, JoinType.LEFT);
-            answerFetch.fetch(AnswerPost_.REACTIONS, JoinType.LEFT);
-            root.fetch(Post_.REACTIONS, JoinType.LEFT);
-            root.fetch(Post_.TAGS, JoinType.LEFT);
-            return criteriaBuilder.equal(root.get(Post_.CONVERSATION).get(Conversation_.ID), conversationId);
-        });
+        return ((root, query, criteriaBuilder) -> criteriaBuilder.equal(root.get(Post_.CONVERSATION).get(Conversation_.ID), conversationId));
```

Comment:
doesn't this change the logic of the method? it seems that reactions, answers and tags are not loaded anymore

Reply:
No, because `Post` already fetches these eagerly by its entity definition. Spring Boot 3 doesn't like duplicate join (there are joins in another method) hence I simply removed it as they get loaded (I also tested it).